views/js/pciCreator/ims/mathEntryInteraction/runtime/mathEntryInteraction.js

```diff
@@ -101,26 +101,62 @@ define([
         };
     });
 
-    var mathEntryInteractionFactory = function() {
+    const labels = {
+        'ja': {
+            'x/y': '<span>x</span><br><span style="text-decoration: overline;">y</span>',
+            '&le;': '&#8806;',
+            '&ge;': '&#8807;',
+            '\\le':  '\\leq',
+            '\\ge':  '\\geq'
+        }
+    };
+
+    var mathEntryInteractionFactory = function () {
         return {
 
             /**
              * return {Boolean} - Are we in a TAO QTI Creator context?
              */
             inQtiCreator: function isInCreator() {
                 if (_.isUndefined(this._inQtiCreator) && this.$container) {
-                    this._inQtiCreator = this.$container.find('.mathEntryInteraction').hasClass('tao-qti-creator-context');
+                    this._inQtiCreator = this.$container.hasClass('tao-qti-creator-context') ||
+                        this.$container.find('.tao-qti-creator-context').length > 0;
                 }
                 return this._inQtiCreator;
             },
 
+            /**
+             * @returns {string} - Localazed label
+             */
+             getLabel: function getLabel(label) {
+                var locale = labels[this.config.locale];
+                if (locale) {
+                    return locale[label] || label;
+                }
+                return label;
+            },
+
             /**
              * @returns {Boolean} - Is the PCI instance configured to use gap expressions?
              */
-            inGapMode: function inGapMode() {
+             inGapMode: function inGapMode() {
                 return this.config.useGapExpression;
             },
 
+            /**
+             * @returns {Boolean} - Is the PCI instance using japanese locale?
+             */
+            inJapanese: function inJapanese() {
+                 return this.userLanguage === 'ja';
```

Comment:
We can do that, but then we will have to create label object for each possible option of japanese localized labels. I think what we have now is an optimal solution. I will add a case where localized labels can be split by `_` though.

Reply:
I think this can be a source of issues. When running the interaction with TAO you'll receive `ja` but the test can also be launched using LTI where the language code will be set by an external system. It can be `ja`, `JA`, `ja-JP`, `ja_JP`, `ja_JP.LEUK` or `ja_HE`, etc. 

You can extract the lang code and rely on it:
```js
 var language = this.userLanguage ? this.userLanguage.replace(/[-_][A-Z].*$/i, '').toLowerCase() : '';

            inJapanese: function inJapanese() {
                 return language === 'ja';
            }
```

What do you think

Reply:
Yeah, I think regular expression can get the job done as well, I will change it